implementations/rust/ockam/ockam_transport_tcp/src/transport/lifecycle.rs

```diff
@@ -51,19 +51,25 @@ impl Transport for TcpTransport {
     }
 
     async fn resolve_route(&self, flow_controls: &FlowControls, route: Route) -> Result<Route> {
-        let mut result = Route::new();
         let mut number_of_tcp_hops = 0;
-
-        for address in route.iter() {
+        for address in route.clone().iter() {
```

Comment:
Is `.clone()` require here?

Reply:
Also, you should be able to do something like
`route.iter().filter(...).count() > 1`